### PR TITLE
COR-51 fix(backend): Add proxyUrl prop

### PR DESCRIPTION
**Pull request by raptisj**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [x] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

Add proxyUrl prop and refactor verifyToken function so it can be called with the proxy url if it is present

### CODE REVIEW

File: packages/backend/src/tokens/interstitialRule.ts [hunk 1]
```diff
@@ -125,11 +125,11 @@ export async function runInterstitialRules<T>(opts: T, rules: InterstitialRule[]
   return signedOut<T>(opts, AuthErrorReason.UnexpectedError);
 }
 
-async function verifyRequestState(options: any, token: string) {
-  const issuer = (iss: string) => iss.startsWith('https://clerk.') || iss.includes('.clerk.accounts');
+async function verifyRequestState(options: any, token: string, issuer = '') {
```
Comment (panteliselef) [inline packages/backend/src/tokens/interstitialRule.ts:128]: ❓ Is this relevant to this PR ? Maybe I am missing some context here

File: packages/remix/src/ssr/types.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export type RootAuthLoaderOptions = {
   frontendApi?: string;
   publishableKey?: string;
   jwtKey?: string;
+  proxyUrl?: string;
```
Comment (panteliselef) [inline packages/remix/src/ssr/types.ts:13]: ❓Doesn't this expose the proxyUrl to user when calling `getAuth` & `rootAuthLoader` and passing options ?

**Review by panteliselef — CHANGES_REQUESTED** [review]

File: packages/backend/src/tokens/interstitialRule.ts (hunk 1 above)
Comment (panteliselef) [reply packages/backend/src/tokens/interstitialRule.ts:128]: In the description you are mentioning "refactor verifyToken function"
But is this necessary for this PR ?

Comment (raptisj) [reply packages/backend/src/tokens/interstitialRule.ts:128]: @panteliselef I updated the description. Yes, this is part of this PR. We need to use the proxyUrl if present during verification since it makes a request to FAPI.

Comment (panteliselef) [reply packages/backend/src/tokens/interstitialRule.ts:128]: @raptisj go it

File: packages/remix/src/ssr/types.ts (hunk 2 above)
Comment (raptisj) [reply packages/remix/src/ssr/types.ts:13]: @panteliselef I omitted it in `rootAuthLoader`. In `getAuth` we pick certain values so we are ok

**Review by panteliselef — CHANGES_REQUESTED** [review]
Looks good to me

File: packages/backend/src/tokens/interstitial.ts [hunk 3]
```diff
@@ -67,6 +69,7 @@ export function loadInterstitialFromLocal(options: Omit<LoadInterstitialOptions,
                     ? `script.setAttribute('data-clerk-publishable-key', '${publishableKey}');`
                     : `script.setAttribute('data-clerk-frontend-api', '${frontendApi}');`
                 }
+                script.setAttribute('data-clerk-proxy-url', '${proxyUrl}');
```
Comment (panteliselef) [inline packages/backend/src/tokens/interstitial.ts:72]: ❓ Should this line be added if proxyUrl is undefined ?

File: packages/backend/src/tokens/interstitialRule.ts [hunk 4]
```diff
@@ -126,11 +126,11 @@ export async function runInterstitialRules<T>(opts: T, rules: InterstitialRule[]
...
```
Comment (SokratisVidros) [inline packages/backend/src/tokens/interstitialRule.ts:129]: ⛔ My recommendation is to avoid changing the signature of `verifyRequestState` method and pass the issuer as another key in the options. The options argument was introduced to cater for such cases.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: packages/backend/src/tokens/interstitialRule.ts (hunk 4 above)
Comment (raptisj) [reply packages/backend/src/tokens/interstitialRule.ts:129]: @SokratisVidros so you recommend removing the `issuer` param from `verifyRequestState` and using the `option.proxyurl` straight away right?

Comment (SokratisVidros) [reply packages/backend/src/tokens/interstitialRule.ts:129]: Yes
